llvm/lib/Target/DirectX/DXILLegalizePass.cpp

```diff
@@ -20,31 +15,27 @@
 #include "llvm/Pass.h"
 #include "llvm/Transforms/Utils/BasicBlockUtils.h"
 #include <functional>
-#include <map>
-#include <stack>
-#include <vector>
 
 #define DEBUG_TYPE "dxil-legalize"
 
 using namespace llvm;
 namespace {
 
-static void fixI8TruncUseChain(Instruction &I,
-                               std::stack<Instruction *> &ToRemove,
-                               std::map<Value *, Value *> &ReplacedValues) {
+void fixI8TruncUseChain(Instruction &I, SmallVector<Instruction *> &ToRemove,
```

Comment:
I think you missed the intent here. These functions should be static, the namespace should be restricted to just class definitions:

> ...we have a simple guideline: make anonymous namespaces as small as possible, and only use them for class declarations.

source: https://llvm.org/docs/CodingStandards.html#restrict-visibility